main.cpp

```diff
@@ -70,29 +70,412 @@ write 3 UDTs below that EACH have:
+    storageInGb -= numFiles * fileSize;
+    std::cout << "Computer::storeData() " << database << std::endl;
+}
+
+void Computer::processData(float amountToProcess, float timeToProcess)
+{
+    processorSpeed = amountToProcess * timeToProcess;
+    std::cout << "Computer::processData() speed: " << processorSpeed << std::endl;
+}
+
+void Computer::displayData(std::string fileName, bool fullscreen)
+{
+    if (!fullscreen)
+        displaySize *= 0.5f;
+
+    std::string displayText = "file contains: " + fileName;    
+    std::cout << "Computer::displayData(): " << displayText << std::endl;
+}
+
+void Computer::changeDisplayBrightness(int newBrightVal)
+{
+    if(newBrightVal > displayBrightness)
+    {
+        for(; displayBrightness < newBrightVal; ++displayBrightness)
+        {
+            std::cout << "increasing brightness: " << displayBrightness << std::endl;
+        }
+    }
+    else
+    {
+        for(; displayBrightness > newBrightVal; --displayBrightness)
+        {
+            std::cout << "decreasing brightness: " << displayBrightness << std::endl;
+        }
+    }  
+}
 /*
  new UDT 4:
  with 2 member functions
  */
+struct CoffeeShop
+{
+    CoffeeShop();
+    ~CoffeeShop();
+
+    CoffeeMachine coffeeMachine1;
+
+    void prepareOrder(int cupsOrdered, int brewStrength);
+    void cleanMachine(int cyclesMade);
 
+};
+
+CoffeeShop::CoffeeShop()
+{
+    std::cout << "CoffeeShop being constructed" << std::endl;
+}
+
+CoffeeShop::~CoffeeShop()
+{
+    std::cout << "CoffeeShop destroyed" << std::endl;
+}
+
+void CoffeeShop::cleanMachine(int cyclesMade)
+{
+    if(coffeeMachine1.brewSettings.flashDescalingIndicator(3, cyclesMade))
+    {
+        std::cout << "starting descaling program for " << coffeeMachine1.brandName << " machine 1 ..." << std::endl;
+        std::cout << "resetting cycle counter... program finished!" << std::endl;
+    }
+    else
+        return;
+}
+
+void CoffeeShop::prepareOrder(int cupsOrdered, int orderedStrength)
+{
+    coffeeMachine1.brewSettings.brewStrength = orderedStrength;
+    coffeeMachine1.brewSettings.brewing(cupsOrdered, 0);
+}
 /*
  new UDT 5:
  with 2 member functions
  */
+struct TransportTracking
+{
+    TransportTracking();
+    ~TransportTracking();
 
+    CargoShip transport;
+    Computer computer;
+
+    void transportStatus(int loaded, int toLoad);
+    void displayTransportItems();
+};
+
+TransportTracking::TransportTracking()
+{
+    std::cout << "TransportTracking being constructed" << std::endl;
+}
+
+TransportTracking::~TransportTracking()
+{
+    std::cout << "TransportTracking destroyed" << std::endl;
+}
+
+void TransportTracking::transportStatus(int loaded, int toLoad)
+{
+    if(transport.readyForDeparture(loaded, toLoad))
+    {
+        computer.storeData(1, 24, "registered transport");
+        std::cout << "tracking status: Delivery is on its way" << std::endl;
+    }
+    else
+        std::cout << "tracking status: transport still in process..." << std::endl;
+}
```

Comment:
If you use curly braces in an `if` statement, use them in the `else` statement.
- [ ] revise